mypy/plugin.py

```diff
@@ -263,17 +264,26 @@ def typed_dict_get_signature_callback(ctx: MethodSigContext) -> CallableType:
             and len(ctx.args[0]) == 1
             and isinstance(ctx.args[0][0], StrExpr)
             and len(signature.arg_types) == 2
-            and len(signature.variables) == 1):
+            and len(signature.variables) == 1
+            and len(ctx.args[1]) == 1):
         key = ctx.args[0][0].value
         value_type = ctx.type.items.get(key)
+        ret_type = signature.ret_type
         if value_type:
+            default_arg = ctx.args[1][0]
+            if (isinstance(value_type, TypedDictType)
+                    and isinstance(default_arg, DictExpr)
+                    and len(default_arg.items) == 0):
+                # Caller has empty dict {} as default for typed dict.
```

Comment:
We special this so that the context will be a non-total TypedDict in case the default is `{}`. If we don't do that, `{}` won't be accepted for total TypedDicts since it has missing keys (all keys are missing).

Any subset of keys should work but this will be harder to implement and likely not very common, so I think it's okay to postpone it until later. I can create an issue to track that.